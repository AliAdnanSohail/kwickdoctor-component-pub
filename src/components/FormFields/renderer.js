import React from 'react';
import classnames from 'classnames';

import styles, { error as errorStyles, label as labelStyles } from './styles';

export default render => ({
  input, meta, label, ...rest
}) => (
  <div className={classnames('form-field', rest.containerClassName)}>
    <label htmlFor={rest.id}>{label}</label>
    {render(input, label, {
      ...rest,
      className: classnames({
        invalid: meta && meta.error && meta.touched,
        active: meta && meta.active,
      }),
    })}
    {meta && meta.error && meta.touched ? <div className="error">{meta.error}</div> : undefined}

    <style jsx>{styles}</style>
    <style jsx>{labelStyles}</style>
    <style jsx>{errorStyles}</style>
  </div>
);
