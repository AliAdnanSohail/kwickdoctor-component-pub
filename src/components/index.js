export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
