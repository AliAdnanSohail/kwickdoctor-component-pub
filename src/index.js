export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
