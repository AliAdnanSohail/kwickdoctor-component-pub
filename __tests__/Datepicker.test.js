import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';
import { mount } from 'enzyme';

import { Datepicker } from '../src';

const testDate = '2018-03-15';
const baseProps = {
  id: 'datepicker-id',
  name: 'datepicker-name',
  onChange() {},
  value: moment(testDate),
  label: 'datepicker-label',
  error: '',
  dateFormat: 'DD MMM, YYYY',
};

it('Datepicker renders correctly', () => {
  const element = renderer.create(<Datepicker {...baseProps} />).toJSON();

  expect(element).toMatchSnapshot();
});

it('Datepicker set id correctly', () => {
  const element = mount(<Datepicker {...baseProps} />);

  expect(element.find(`[htmlFor="${baseProps.id}"]`)).toHaveLength(1);
  expect(element.find(`input#${baseProps.id}`)).toHaveLength(1);
});

it('Datepicker set name correctly', () => {
  const element = mount(<Datepicker {...baseProps} />);

  expect(element.find(`input[name="${baseProps.name}"]`)).toHaveLength(1);
});

it('Datepicker set value correctly', () => {
  const element = mount(<Datepicker {...baseProps} />);

  expect(element.find('input').instance().value).toEqual(baseProps.value.format(baseProps.dateFormat));
});

it('Datepicker set label correctly', () => {
  const element = mount(<Datepicker {...baseProps} />);

  expect(element.find('label').text()).toEqual(baseProps.label);
});

it('Datepicker input correctly', () => {
  const props = Object.assign({}, baseProps);
  const inputValue = '31 Jan, 2017';
  let value;
  props.onChange = (date) => {
    value = date;
  };

  const element = mount(<Datepicker {...props} />);
  const input = element.find('input');
  input.simulate('focus');
  input.simulate('change', { target: { value: inputValue } });

  expect(inputValue === value.format('DD MMM, YYYY')).toEqual(true);
});

it('Datepicker show error correctly', () => {
  const props = Object.assign({}, baseProps);
  props.error = 'error-text';
  const element = mount(<Datepicker {...props} />);

  expect(element.find('.input-error').text()).toEqual(props.error);
});

it('Datepicker click by day correctly', () => {
  const props = Object.assign({}, baseProps);
  let value = props.value.clone();
  props.onChange = (date) => {
    value = date;
  };
  const element = mount(<Datepicker {...props} />);
  element.find('input').simulate('focus');
  element.find('[aria-label="day-16"]').simulate('click');

  expect(props.value.isSame(value, 'day')).toEqual(false);
});
