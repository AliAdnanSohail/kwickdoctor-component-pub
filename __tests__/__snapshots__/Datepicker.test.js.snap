// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker renders correctly 1`] = `
<div
  className="datepicker-block"
>
  <label
    htmlFor="datepicker-id"
  >
    datepicker-label
  </label>
  <div>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <input
          autoComplete={undefined}
          autoFocus={undefined}
          className="datepicker"
          disabled={false}
          id="datepicker-id"
          name="datepicker-name"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder={undefined}
          readOnly={undefined}
          required={undefined}
          tabIndex={undefined}
          title={undefined}
          type="text"
          value="15 Mar, 2018"
        />
      </div>
    </div>
  </div>
  
  <style>
    .datepicker-block{position:relative;}
  </style>
  <style>
    .datepicker-block .calendar{background-color:#ffffff;border:none;border-radius:6px;box-shadow:0 2px 40px 0 rgba(0,0,0,0.05);}
    .datepicker-block .calendar .react-datepicker__header{border:none;background-color:#ffffff;}
    .datepicker-block .calendar .react-datepicker__month{margin-top:0;}
    .datepicker-block .calendar .react-datepicker__month-container{padding:10px 20px 20px 20px;}
    .datepicker-block .calendar .react-datepicker__current-month{font-size:1rem;margin-bottom:20px;}
    .datepicker-block .calendar .react-datepicker__triangle{position:absolute;top:0 !important;left:36px !important;width:0;height:0;margin:0 !important;-webkit-transform-origin:0 0;-ms-transform-origin:0 0;transform-origin:0 0;-webkit-transform:rotate(135deg);-ms-transform:rotate(135deg);transform:rotate(135deg);box-sizing:border-box !important;box-shadow:-4px 4px 14px 0 rgba(0,0,0,0.05) !important;border-width:6px !important;border-style:solid !important;border-color:transparent transparent #ffffff #ffffff !important;}
    .datepicker-block .calendar .react-datepicker__triangle:before{content:none !important;}
    .datepicker-block .calendar .react-datepicker__navigation{border:none;top:17px;}
    .datepicker-block .calendar .react-datepicker__navigation:before{content:'';display:block;width:24px;height:24px;position:absolute;left:-39px;color:#222328;background-image:url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' version='1.1' viewBox='0 0 24 24' width='24px' height='24px' role='img' class='grommetux-control-icon grommetux-control-icon-form-next grommetux-control-icon--responsive' aria-label='form-next'%3E%3Cpolyline fill='none' stroke='%23000' stroke-width='2' points='9 6 15 12 9 18'%3E%3C/polyline%3E%3C/svg%3E");background-repeat:no-repeat;font-weight:bold;font-size:1.3rem;}
    .datepicker-block .calendar .react-datepicker__navigation--next{right:16px;}
    .datepicker-block .calendar .react-datepicker__navigation--previous{left:16px;}
    .datepicker-block .calendar .react-datepicker__navigation--previous:before{-webkit-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);left:18px;}
    .datepicker-block .calendar .react-datepicker__day{margin:6px;border-radius:50%;width:24px;height:24px;line-height:24px;color:#2A2A30;font-size:0.75rem;font-weight:400;}
    .datepicker-block .calendar .react-datepicker__day:hover:not(.react-datepicker__day--selected){color:#2A2A30;background-color:#eff0f4;}
    .datepicker-block .calendar .react-datepicker__day--selected{color:#ffffff;background-color:#0c97f9;}
    .datepicker-block .calendar .react-datepicker__day--outside-month{color:#bbbccd;}
    .datepicker-block .calendar .react-datepicker__day-name{margin:0 6px;color:#bbbccd;width:24px;height:24px;line-height:24px;text-transform:uppercase;font-size:0.75rem;font-weight:400;}
  </style>
  <style>
    .datepicker-block .datepicker{width:100%;padding:12px 16px 12px 52px;color:#2A2A30;font-size:0.875rem;box-sizing:border-box;background-color:#ffffff;background-repeat:no-repeat;background-position:17px center;background-image:url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='UTF-8'%3F%3E%3Csvg width='19px' height='20px' viewBox='0 0 19 20' version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'%3E%3C!-- Generator: Sketch 48 %2847235%29 - http://www.bohemiancoding.com/sketch --%3E%3Ctitle%3Ecalendar%3C/title%3E%3Cdesc%3ECreated with Sketch.%3C/desc%3E%3Cdefs%3E%3Cpath d='M4,4 C3.44771525,4 3,4.44771525 3,5 L3,17 C3,17.5522847 3.44771525,18 4,18 L17,18 C17.5522847,18 18,17.5522847 18,17 L18,5 C18,4.44771525 17.5522847,4 17,4 L4,4 Z M4,2 L17,2 C18.6568542,2 20,3.34314575 20,5 L20,17 C20,18.6568542 18.6568542,20 17,20 L4,20 C2.34314575,20 1,18.6568542 1,17 L1,5 C1,3.34314575 2.34314575,2 4,2 Z M3,7 L18,7 L18,9 L3,9 L3,7 Z M6,0.5 L6,0.5 C6.55228475,0.5 7,0.94771525 7,1.5 L7,5 C7,5.55228475 6.55228475,6 6,6 L6,6 C5.44771525,6 5,5.55228475 5,5 L5,1.5 C5,0.94771525 5.44771525,0.5 6,0.5 Z M15,0.5 L15,0.5 C15.5522847,0.5 16,0.94771525 16,1.5 L16,5 C16,5.55228475 15.5522847,6 15,6 L15,6 C14.4477153,6 14,5.55228475 14,5 L14,1.5 C14,0.94771525 14.4477153,0.5 15,0.5 Z M15.25,13 C15.9403559,13 16.5,12.4403559 16.5,11.75 C16.5,11.0596441 15.9403559,10.5 15.25,10.5 C14.5596441,10.5 14,11.0596441 14,11.75 C14,12.4403559 14.5596441,13 15.25,13 Z M15.25,16.8000002 C15.9403559,16.8000002 16.5,16.2403561 16.5,15.5500002 C16.5,14.8596443 15.9403559,14.3000002 15.25,14.3000002 C14.5596441,14.3000002 14,14.8596443 14,15.5500002 C14,16.2403561 14.5596441,16.8000002 15.25,16.8000002 Z M10.5500002,13 C11.2403561,13 11.8000002,12.4403559 11.8000002,11.75 C11.8000002,11.0596441 11.2403561,10.5 10.5500002,10.5 C9.85964425,10.5 9.30000019,11.0596441 9.30000019,11.75 C9.30000019,12.4403559 9.85964425,13 10.5500002,13 Z M10.5500002,16.8000002 C11.2403561,16.8000002 11.8000002,16.2403561 11.8000002,15.5500002 C11.8000002,14.8596443 11.2403561,14.3000002 10.5500002,14.3000002 C9.85964425,14.3000002 9.30000019,14.8596443 9.30000019,15.5500002 C9.30000019,16.2403561 9.85964425,16.8000002 10.5500002,16.8000002 Z M5.75,13 C6.44035594,13 7,12.4403559 7,11.75 C7,11.0596441 6.44035594,10.5 5.75,10.5 C5.05964406,10.5 4.5,11.0596441 4.5,11.75 C4.5,12.4403559 5.05964406,13 5.75,13 Z M5.75,16.8000002 C6.44035594,16.8000002 7,16.2403561 7,15.5500002 C7,14.8596443 6.44035594,14.3000002 5.75,14.3000002 C5.05964406,14.3000002 4.5,14.8596443 4.5,15.5500002 C4.5,16.2403561 5.05964406,16.8000002 5.75,16.8000002 Z' id='path-1'%3E%3C/path%3E%3C/defs%3E%3Cg id='Icons/Calendar' stroke='none' stroke-width='1' fill='none' fill-rule='evenodd' transform='translate%28-1.000000, 0.000000%29'%3E%3Cg id='icon'%3E%3Cmask id='mask-2' fill='white'%3E%3Cuse xlink:href='%23path-1'%3E%3C/use%3E%3C/mask%3E%3Cuse id='calendar' fill='%23cccccc' fill-rule='nonzero' xlink:href='%23path-1'%3E%3C/use%3E%3C/g%3E%3C/g%3E%3C/svg%3E");border-width:2px;border-style:solid;border-color:#E7E7F0;border-radius:4px;}
    .datepicker-block .datepicker:focus{outline:none;border-color:#0c97f9;}
    .datepicker-block.has-error .datepicker,.datepicker-block.has-error .datepicker:focus{border-color:#FF5572;}
  </style>
  <style>
    .datepicker-block label{color:#2A2A30;display:block;margin-bottom:6px;font-weight:500;font-size:1rem;}
  </style>
  <style>
    .datepicker-block .react-datepicker-wrapper,.datepicker-block .react-datepicker__input-container{width:100%;}
  </style>
  <style>
    .datepicker-block .input-error{position:absolute;left:4px;width:auto;max-width:100%;box-sizing:border-box;word-wrap:break-word;padding:12px 16px;margin-top:-6px;border-radius:3px;background-color:#ffffff;box-shadow:0 2px 14px 0 rgba(0,0,0,0.14);color:#FF5572;font-size:0.875rem;font-weight:500;}
    .datepicker-block .input-error:before{content:'';position:absolute;top:0;left:32px;width:0;height:0;-webkit-transform-origin:0 0;-ms-transform-origin:0 0;transform-origin:0 0;-webkit-transform:rotate(135deg);-ms-transform:rotate(135deg);transform:rotate(135deg);box-sizing:border-box;box-shadow:-4px 4px 14px 0 rgba(0,0,0,0.05);border-width:6px;border-style:solid;border-color:transparent transparent #ffffff #ffffff;}
  </style>
</div>
`;
